streams/shared/src/main/scala/zio/stream/internal/ChannelExecutor.scala

```diff
@@ -448,21 +452,21 @@ private[zio] final class ChannelExecutor[Env, InErr, InElem, InDone, OutErr, Out
   private[this] def runFinalizers(finalizers: Iterable[Finalizer[Env]], ex: Exit[Any, Any])(implicit
     trace: Trace
   ): URIO[Env, Any] =
-    if (finalizers.isEmpty) null
+    if (finalizers.isEmpty) ZIO.unit
```

Comment:
_Famous last words_

I think it'd rather have `null` returned here. The reason is that it's much more likely for a bug that would degrade performance to be caught if we had `null` as we'd get an NPE exception during CI. But since `ZIO.unit` will work, it might degrade performance especially if we end up flatmapping a lot of them unnecessarily